File: backend/app/schemas/sales.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel
from decimal import Decimal


# Customer schemas
class CustomerCreate(BaseModel):
    name: str
    phone: str
    province: Optional[str] = "江苏省"
    city: Optional[str] = "苏州市"
    district: Optional[str] = "吴中区"
    town: Optional[str] = "临湖镇"
    address: Optional[str] = None
    contact: Optional[str] = None
    remark: Optional[str] = None


class CustomerUpdate(BaseModel):
    name: Optional[str] = None
    phone: Optional[str] = None
    province: Optional[str] = None
    city: Optional[str] = None
    district: Optional[str] = None
    town: Optional[str] = None
    address: Optional[str] = None
    contact: Optional[str] = None
    remark: Optional[str] = None


class CustomerResponse(BaseModel):
    id: int
    name: str
    phone: str
    province: str
    city: str
    district: str
    town: str
    address: Optional[str] = None
    contact: Optional[str] = None
    remark: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


# Sales Order Item schemas
class SalesOrderItemCreate(BaseModel):
    product_id: int
    quantity: int
    unit_price: Decimal


class SalesOrderItemResponse(BaseModel):
    id: int
    order_id: int
    product_id: int
    product_name: Optional[str] = None
    product_spec: Optional[str] = None
    product_unit: Optional[str] = None
    quantity: int
    unit_price: Decimal
    subtotal: Decimal

    model_config = {"from_attributes": True}


# Old Appliance schema (for trade-in)
class OldApplianceCreate(BaseModel):
    category: str
    brand: Optional[str] = None
    condition: str = "旧"
    recycle_price: Decimal = Decimal("0")
    warehouse_id: Optional[int] = None
    remark: Optional[str] = None


class OldApplianceResponse(BaseModel):
    id: int
    category: str
    brand: Optional[str] = None
    condition: str
    recycle_price: Decimal
    warehouse_id: Optional[int] = None
    recycle_date: datetime
    remark: Optional[str] = None

    model_config = {"from_attributes": True}


# Sales Order schemas
class SalesOrderCreate(BaseModel):
    customer_id: int
    items: List[SalesOrderItemCreate]
    discount_amount: Optional[Decimal] = Decimal("0")
    old_appliances: Optional[List[OldApplianceCreate]] = None
    remark: Optional[str] = None


class SalesOrderUpdate(BaseModel):
    customer_id: Optional[int] = None
    items: Optional[List[SalesOrderItemCreate]] = None
    discount_amount: Optional[Decimal] = None
    payment_status: Optional[str] = None
    remark: Optional[str] = None


class SalesOrderResponse(BaseModel):
    id: int
    order_no: str
    customer_id: int
    customer_name: Optional[str] = None
    customer_phone: Optional[str] = None
    customer_address: Optional[str] = None
    total_amount: Decimal
    discount_amount: Decimal
    final_amount: Decimal
    payment_status: str
    status: str
    remark: Optional[str] = None
    created_at: datetime
    items: List[SalesOrderItemResponse] = []
    old_appliances: List[OldApplianceResponse] = []

    model_config = {"from_attributes": True}

File: frontend/src/types/sales.ts
// 客户相关类型
export interface Customer {
  id: number;
  name: string;
  phone: string;
  province: string;
  city: string;
  district: string;
  town: string;
  address?: string;
  contact?: string;
  remark?: string;
  created_at: string;
}

export interface CustomerCreate {
  name: string;
  phone: string;
  province?: string;
  city?: string;
  district?: string;
  town?: string;
  address?: string;
  contact?: string;
  remark?: string;
}

export interface CustomerUpdate {
  name?: string;
  phone?: string;
  province?: string;
  city?: string;
  district?: string;
  town?: string;
  address?: string;
  contact?: string;
  remark?: string;
}

// 销售单明细
export interface SalesOrderItem {
  id: number;
  order_id: number;
  product_id: number;
  product_name?: string;
  product_spec?: string;
  product_unit?: string;
  quantity: number;
  unit_price: number;
  subtotal: number;
  warehouses?: { warehouse_id: number; warehouse_name: string; quantity: number }[];
}

export interface SalesOrderItemCreate {
  product_id: number;
  quantity: number;
  unit_price: number;
}

// 旧电器（以旧换新）
export interface OldAppliance {
  id: number;
  category: string;
  brand?: string;
  condition: string;
  recycle_price: number;
  warehouse_id?: number;
  recycle_date: string;
  remark?: string;
}

export interface OldApplianceCreate {
  category: string;
  brand?: string;
  condition?: string;
  recycle_price?: number;
  warehouse_id?: number;
  remark?: string;
}

// 销售单
export interface SalesOrder {
  id: number;
  order_no: string;
  customer_id: number;
  customer_name?: string;
  customer_phone?: string;
  customer_address?: string;
  total_amount: number;
  discount_amount: number;
  subsidy_amount?: number;
  final_amount: number;
  payment_status: string;
  paid_amount?: number;
  status: string;
  remark?: string;
  created_by?: number;
  created_by_name?: string;
  cancel_reason?: string;
  cancelled_at?: string;
  cancelled_by_name?: string;
  created_at: string;
  items: SalesOrderItem[];
  old_appliances: OldAppliance[];
  payments?: OrderPayment[];
}

export interface SalesOrderCreate {
  customer_id: number;
  items: SalesOrderItemCreate[];
  discount_amount?: number;
  subsidy_amount?: number;
  old_appliances?: OldApplianceCreate[];
  remark?: string;
}

export interface SalesOrderUpdate {
  customer_id?: number;
  items?: SalesOrderItemCreate[];
  discount_amount?: number;
  subsidy_amount?: number;
  payment_status?: string;
  remark?: string;
}

// 支付方式
export const PAYMENT_METHODS = [
  { value: '现金', label: '现金' },
  { value: '数字人民币', label: '数字人民币' },
  { value: '微信', label: '微信' },
  { value: '支付宝', label: '支付宝' },
  { value: '信用卡', label: '信用卡' },
  { value: '银行转账', label: '银行转账' },
  { value: '其他', label: '其他' },
] as const

export interface OrderPayment {
  id: number
  order_id: number
  payment_method: string
  amount: number
  remark?: string
  created_at: string
  created_by_name?: string
}

export interface OrderPaymentCreate {
  payment_method: string
  amount: number
  remark?: string
}